### PR TITLE
Dispose tooltips

### PR DESCRIPTION
**Pull request by PaulBol**
I noticed that some controls do not dispose tooltips and timers. Depending on the underlying implementation these may use native resources. Therefore I'd suggest to dispose these with the parent control to not rely on the garbage collector to eventually release the resources.

Additionally, in the case of `ToolStrip.cs` `CloseToolTip` would first create the objects before turning them off.

### CODE REVIEW

File: mcs/class/System.Windows.Forms/System.Windows.Forms/ToolStrip.cs [hunk 1]
```diff
@@ -743,6 +742,15 @@ protected override void Dispose (bool disposing)
 						this.overflow_button.drop_down.Dispose ();
 
 					ToolStripManager.RemoveToolStrip (this);
+
+					if (tooltip_timer != null)
+					{
+						tooltip_timer.Stop();
+						tooltip_timer.Dispose();
```
Comment (filipnavara) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/ToolStrip.cs]: Is there any reason why you sometime call just Dispose and other times you call Stop and then Dispose? I would expect the Dispose to be sufficient.

Comment (PaulBol) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/ToolStrip.cs]: That's true. `Dispose` includes `Stop`. In `ToolStrip.cs` I copied the `Stop` from `CloseToolTip` which previously was called. Let me try to update the PR.

**Comment by lewurm** [comment]
@monojenkins build

**Comment by lewurm** [comment]
@monojenkins approve

**Comment by lewurm** [comment]
@monojenkins build

**Comment by akoeplinger** [comment]
@monojenkins build

**Comment by akoeplinger** [comment]
Thank you for the contribution!

**Pull request merged**